geoportailv3/static/js/usermanagerservice.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * @fileoverview this file defines the user managerservice. this service
+ * interacts with the Geoportail webservice to login logout and keep the state
+ * of the user
+ */
+
+goog.provide('app.UserManager');
+
+goog.require('app');
+goog.require('app.Notify');
+
+
+
+/**
+ * @constructor
+ * @param {angular.$http} $http Angular http service.
+ * @param {string} loginUrl The application login URL.
+ * @param {string} logoutUrl The application logout URL.
+ * @param {string} getuserinfoUrl The url to get information about the user.
+ * @param {app.Notify} appNotify Notify service.
+ * @param {gettext} gettext Gettext service.
+ * @ngInject
+ */
+app.UserManager = function($http, loginUrl, logoutUrl,
+    getuserinfoUrl, appNotify, gettext) {
+  /**
+   * @type {string}
+   * @private
+   */
+  this.loginUrl_ = loginUrl;
+
+  /**
+   * @type {app.Notify}
+   * @private
+   */
+  this.notify_ = appNotify;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.getuserinfoUrl_ = getuserinfoUrl;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.logoutUrl_ = logoutUrl;
+
+  /**
+   * @type {Object}
+   * @private
+   */
+  this['credentials'] = {
+    'login' : null,
+    'password' : null
+  };
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+
+  /**
+   * @type {string|undefined}
+   */
+  this['login'] = undefined;
```

Comment:
Bottom line is: don't use strings (and the bracket notation) when it's not strictly necessary, as using strings prevent renaming. Use `strings` (or `@export`) in directive controllers when properties are used in templates.